### PR TITLE
fix: Fix table alias used on count() query

### PR DESCRIPTION
**Pull request by hectoras**
**Associated Jia issue:** [ADF-1054](https://oat-sa.atlassian.net/browse/ADF-1054).

### CODE REVIEW

**Review by shpran — APPROVED** [review]

File: models/classes/Lists/DataAccess/Repository/RdsValueCollectionRepository.php [hunk 1]
```diff
@@ -184,7 +184,7 @@ public function count(ValueCollectionSearchRequest $searchRequest): int
         $this->enrichQueryWithInitialCondition($query);
         $this->enrichQueryWithValueCollectionSearchCondition($searchRequest, $query);
 
-        $result = $query->select('count(t.id) AS c')->execute();
+        $result = $query->select('count(items.id) AS c')->execute();
```
Comment (gabrielfs7) [inline models/classes/Lists/DataAccess/Repository/RdsValueCollectionRepository.php:187]: - [ ] We just need to validate if the test is still passing :)

Comment (hectoras) [reply models/classes/Lists/DataAccess/Repository/RdsValueCollectionRepository.php:187]: They are.

```
~/adf-1054/tao (fix/ADF-1054-fix-count-query-alias)$ ../vendor/bin/phpunit -v \
                                                     --bootstrap ../generis/test/bootstrap.php \
                                                     ./test/unit/models/classes/Lists/
PHPUnit 8.5.23-7-gddf8460a4 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.27 with Xdebug 3.1.2

...............................................................  63 / 189 ( 33%)
............................................................... 126 / 189 ( 66%)
............................................................... 189 / 189 (100%)


Time: 1.94 seconds, Memory: 20.00 MB

OK (189 tests, 630 assertions)
```

However we had previous warnings related with type casting precedence for `Rd*f*ValueCollectionRepository`.

```
1) oat\tao\test\unit\model\Lists\DataAccess\Repository\RdfValueCollectionRepositoryTest::testCount with data set "An empty result set is handled gracefully" (0, array(), 'http://value.collection/', array('http://www.w3.org/2000/01/rdf...#label', 'http://www.w3.org/1999/02/22-...s#type', 'http://value.collection/'))
Undefined offset: 0

2) oat\tao\test\unit\model\Lists\DataAccess\Repository\RdfValueCollectionRepositoryTest::testCount with data set "A null result set is handled gracefully" (0, null, 'http://value.collection/', array('http://www.w3.org/2000/01/rdf...#label', 'http://www.w3.org/1999/02/22-...s#type', 'http://value.collection/'))
Trying to access array offset on value of type null
```

I've committed a fix for that (replicating the fix in this other repo) as part of this PR: f231e3edd8d3dffe869c5ea036004b069f9e2084

**Comment by AntonNeviarouski** [comment]
QA verified

**Comment by hectoras** [comment]
Released as [v50.11.1](https://github.com/oat-sa/tao-core/releases/tag/v50.11.1).

**Comment by hectoras** [comment]
Backported as [v50.10.11.1](https://github.com/oat-sa/tao-core/releases/tag/v50.10.11.1).

**Comment by hectoras** [comment]
Backported as TAO Community release [2022.05.1](https://github.com/oat-sa/tao-community/releases/tag/2022.05.1).

**Pull request merged**